### PR TITLE
CMA microbenchmark subsection

### PR DESCRIPTION
**Pull request by samcusworth**

### CODE REVIEW

**Comment by christophermaynard** [comment]
The rest of the report is written in the impersonal voice. Please use 
"The CMA_appinverse microbenchmark is introduced".
rather than
"we will introduce the CMA_appinverse ...".

**Comment by samcusworth** [comment]
> The rest of the report is written in the impersonal voice. Please use
> "The CMA_appinverse microbenchmark is introduced".
> rather than
> "we will introduce the CMA_appinverse ...".

Good point, I've reworded some of the text... sorry: some of the text has been reworded :-)

File: PA-cma_microbenchmark.tex [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+This kernel is used in the pre-conditioner for the Helmholtz solver to solve the tridiagonal case of equation \ref{eq:matvec} using the Thomas Algorithm~\cite{numericalrecipes}.
+As per the LMA microbenchmark, the driver code was extracted from the Psyclone-generated Psy-layer and the kernel was taken directly from the LFRic source code.
+By design, the driver routine contains the so called global optimisations, namely OpenMP worksharing and colouring (to prevent data race conditions) over columns on the sphere.
+
+The dinodump capability from PYSKE was used to generate Known Good Output (KGO) which we use to verify the microbenchmark behaved as expected in later tests.
+This involved adding additional PSYKE subroutine calls to the Psy-layer code in the gravity\_wave mini-app in order to capture the input scalars and arrays for the kernel.
+The microbenchmark driver code was also trivially modified to read in the produced dinodump before calling the kernel and compare the result of the kernel against the KGO.
```
Comment (christophermaynard) [inline PA-cma_microbenchmark.tex:25]: Dinodumo should be dinodump

File: PA-cma_microbenchmark.tex [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+This kernel is used in the pre-conditioner for the Helmholtz solver to solve the tridiagonal case of equation \ref{eq:matvec} using the Thomas Algorithm~\cite{numericalrecipes}.
+As per the LMA microbenchmark, the driver code was extracted from the Psyclone-generated Psy-layer and the kernel was taken directly from the LFRic source code.
+By design, the driver routine contains the so called global optimisations, namely OpenMP worksharing and colouring (to prevent data race conditions) over columns on the sphere.
+
+The dinodump capability from PYSKE was used to generate Known Good Output (KGO) which was use to verify the microbenchmark behaved as expected in later tests.
+This involved adding additional PSYKE subroutine calls to the Psy-layer code in the gravity\_wave mini-app in order to capture the input scalars and arrays for the kernel.
+The microbenchmark driver code was also trivially modified to read in the produced dinodump before calling the kernel and compare the result of the kernel against the KGO.
+Before running the mini-app to generate the dinodumo the number of vertical layers in the configuration was modified from the default (10) to 256.
```
Comment (samcusworth) [inline PA-cma_microbenchmark.tex]:
```suggestion
Before running the mini-app to generate the dinodump the number of vertical layers in the configuration was modified from the default (10) to 256.
```

File: PA-cma_microbenchmark.tex (hunk 1 above)
Comment (samcusworth) [reply PA-cma_microbenchmark.tex:25]: Whoops. Done (in the web gui!) in commit 64726a5

**Comment by christophermaynard** [comment]
Great. Thanks Sam.

**Pull request merged**